Tests/SentryTests/TestUtils/Async.swift

```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+func delayNonBlocking(timeout: Double = 0.2) {
+    let group = DispatchGroup()
+    group.enter()
+    let queue = DispatchQueue(label: "delay", qos: .background, attributes: [])
+    
+    queue.asyncAfter(deadline: .now() + timeout) {
+        group.leave()
+    }
+    
+    group.wait()
```

Comment:
L:  I know we're just moving something but isn't this blocking? Why is the function name `NonBlocking`?

[Docs on dispatch group:](https://developer.apple.com/documentation/dispatch/1452794-dispatch_group_wait)
> Waits synchronously for the previously submitted block objects to finish; returns if the blocks do not complete before the specified timeout period has elapsed.

Reply:
It's not blocking the thread. I take this as a sign to rename it :)

Reply:
it is blocking the current thread till the previous job gets done or timed out, or `wait` means something else here? I also got this from the docs.

Reply:
Yeah, it's also my understanding that it's blocking. The docs say `synchronously`.
I don't know swift but by reading the code, there's no coroutine or `async/await` that I can spot there so I fail to understand how it would release control and have the code _after the wait_ call be a continuation.

Just trying to understand how it works more than anything. Could be renamed if the name is in fact misleading